tools/gen-magic-strings.py

```diff
@@ -0,0 +1,240 @@
+#!/usr/bin/env python3
+
+# Copyright JS Foundation and other contributors, http://js.foundation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import configparser
+import json
+import os
+import re
+
+from settings import PROJECT_DIR
+
+
+def read_magic_string_definitions():
+    ini_parser = configparser.ConfigParser()
+    ini_parser.optionxform = str # case sensitive options (magic string IDs)
+    ini_parser.read(os.path.join(PROJECT_DIR,
+                                 'jerry-core',
+                                 'lit',
+                                 'lit-magic-strings.ini'))
+
+    defs = [(str_ref, json.loads(str_value))
+            for str_ref, str_value in ini_parser.items('LIT_MAGIC_STRINGS')]
+    defs = sorted(sorted(defs, key=lambda ref_value: ref_value[1]),
+                  key=lambda ref_value: len(ref_value[1]))
+    return defs
+
+
+def extract_magic_string_references():
+    results = dict()
+
+    def process_line(fname, lnum, line, guard_stack):
```

Comment:
I've kept these. They helped me structure the code better without exposing local code to the global scope.